guppylang/diagnostic.py

```
@@ -387,6 +387,104 @@ def level_str(level: DiagnosticLevel) -> str:
         return level.name.lower().capitalize()
 
 
+class MietteRenderer:
+    """Drop-in replacement for DiagnosticsRenderer using miette."""
+
+    def __init__(self, source: SourceMap) -> None:
+        self.buffer: list[str] = []
+        self.source = source
+        try:
+            from miette_py import guppy_to_miette, render_report
+
+            self._guppy_to_miette: Callable[..., Any] = guppy_to_miette
+            self._render_report: Callable[[Any], str] = render_report
+        except ImportError:
+            raise ImportError(
+                "miette-py not available. Install with: pip install miette-py/"
+            ) from None
+
+    def render_diagnostic(self, diag: Diagnostic) -> None:
+        """Renders diagnostic using miette. Same interface as DiagnosticsRenderer."""
+        spans = []
+        source_text = None
+
+        if diag.span:
+            main_span = to_span(diag.span)
+
+            try:
+                context_lines = 10
+                source_lines = self.source.span_lines(main_span, context_lines)
+                source_text = "\n".join(source_lines)
+            except Exception:  # noqa: BLE001
+                source_text = None
```

Please remove this try catch. We should fail if we don't have a source text instead of continuing and then failing later

---

## Response to: "Please remove this try catch. We should fail if we don't have a source text instead of continuing and then failing later"

**Issue Addressed**: Removed the problematic try-catch block that was hiding real errors and creating inconsistent state.

**Solution**: Replaced `span_lines()` call with direct access to the complete file source, eliminating the need for error handling at this level.

### Before:
```python
try:
    context_lines = 10
    source_lines = self.source.span_lines(main_span, context_lines)
    source_text = "\n".join(source_lines)
except Exception:  # noqa: BLE001
    source_text = None
```

### After:
```python
# Get entire file directly from sources
full_file_lines = self.source.sources[main_span.file]
source_text = "\n".join(full_file_lines)
```

**Result**: The code now fails fast and explicitly when there are actual issues, rather than silently continuing with `None` values that cause problems downstream.

---